### ISSUE TITLE
Coding style guidelines

### ISSUE THREAD

**Issue by paulfantom** [prometheus-operator/prometheus-operator#3524]
**What is missing?**

We had a dead link to old coreos coding guidelines, now after https://github.com/prometheus-operator/prometheus-operator/pull/3519 was merged we don't have any coding guidelines.

**Anything else we need to know?**:
This is a follow-up to discussion from https://github.com/prometheus-operator/prometheus-operator/pull/3519#discussion_r492144248

/cc @s-urbaniak

**Comment by lilic** [comment]
I like the https://github.com/thanos-io/thanos/blob/master/docs/contributing/coding-style-guide.md that was suggested! 👍

**Comment by paulfantom** [comment]
We cannot directly use that guideline, unless we will to refactor some part of codebase to use [runutil](https://pkg.go.dev/github.com/thanos-io/thanos@v0.11.0/pkg/runutil?tab=doc)

**Comment by simonpasquier** [comment]
Another data point from https://github.com/prometheus/prometheus/blob/master/CONTRIBUTING.md

>Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments)  and the _Formatting and style_ section of Peter Bourgon's [Go: Best Practices for Production   Environments](https://peter.bourgon.org/go-in-production/#formatting-and-style).

**Comment by s-urbaniak** [comment]
@paulfantom what would speak against refactoring? I would assume that would be a good opportunity to readjust.

**Comment by paulfantom** [comment]
>  what would speak against refactoring

Nothing, apart from time :)

**Comment by brancz** [comment]
It's rather uncommon that there are 0 violations of style guides. I think it's better to have it and improve than adopt a standard we're not actually happy with.

**Comment by s-urbaniak** [comment]
agreed :+1: we could merge the coding styleguides in conjunction with creating a "help needed" issue asking the community to help in refactors to adhere to guidelines.

**Comment by lilic** [comment]
Yes, we could mark them as hacktoberfest label as well, as they are most likely very small issues that are straightforward.
